module/zfs/qat_compress.c

```diff
@@ -468,6 +480,12 @@ qat_compress_impl(qat_compress_dir_t dir, char *src, int src_len,
 			goto fail;
 		}
 
+		/* verify adler checksum */
```

Comment:
Could you explain why this is necessary.  Extra sanity checks are great, but are there specific conditions which we're aware of where this might be an issue?  For example a specific driver or kernel versions?

Reply:
I think the software implementation has done the same check in the gzip/zlib library. @cfzhu can you please confirm?

Reply:
In the function 'inflate()',  the software implementation has done this.
https://github.com/madler/zlib/blob/master/inflate.c#L1197-L1218